### PR TITLE
Maniacs Patch - Call System Functions

### PR DESCRIPTION
**Pull request by jetrotal**
In Maniacs, CommandOpenSaveMenu is renamed to "Call System Function". I implemented what is closer to our code.

Also expanded the code to cover our custom submenus (Cases 8 to 12).

### CODE REVIEW

File: src/game_interpreter_map.cpp [hunk 1]
```diff
@@ -712,15 +713,63 @@ bool Game_Interpreter_Map::CommandPlayMovie(lcf::rpg::EventCommand const& com) {
...
+		return CommandOpenVideoOptions(com);
+	case 5: // Debug menu
+		Scene::instance->SetRequestedScene(std::make_shared<Scene_Debug>());
+		break;
+	case 6: // License information menu
+		// TODO? Implement license information menu
+		return true;
+	case 7: // Reset game
+		return CommandReturnToTitleScreen(com);
+	case 8: // EASYRPG Inventory menu
```
Comment (Ghabry) [inline src/game_interpreter_map.cpp:759]: Is it intentional that an enabled direct menu patch can redirect these menu calls? (the dmpatch redirection is in ``RequestMainMenuScene``)

**Review by Ghabry — COMMENTED** [review]
.

File: src/game_interpreter_map.cpp [hunk 2]
```diff
@@ -712,15 +713,63 @@ bool Game_Interpreter_Map::CommandPlayMovie(lcf::rpg::EventCommand const& com) {
...
+	// Handle save menu (default behavior)
+	if (!Player::IsPatchManiac() || current_system_function <= 0) {
+		Scene::instance->SetRequestedScene(std::make_shared<Scene_Save>());
+		++index;
+		return false;
+	}
+
+	// Parse common parameters
+	const int fullscreen_mode = com.parameters[1]; // Broken in Maniac.
+	const int pause_while_debugging = com.parameters[1]; // unused in our ingame debug screen.
```
[the remark this answers is not in the document]
Comment (jetrotal) [reply src/game_interpreter_map.cpp:735]: Maniacs reuses parameter[1] in different cases, I just game names to the parameters to be easier to read and understand.

**Comment by Ghabry** [comment]
I will move the EasyRPG extension stuff to a higher ID like 200 onwards to prevent conflicts with upcoming MP version. Otherwise this is fine imo.

**Review by fdelapena — APPROVED** [review]

**Pull request merged**